tools/autograd/gen_annotated_fn_args.py

```diff
@@ -13,65 +13,69 @@
 torch/testing/_internal/generated
 """
 
-from .utils import write, CodeTemplate
-from .gen_python_functions import (
-    get_py_nn_functions,
-    get_py_torch_functions,
-    get_py_variable_methods,
-    op_name,
-)
+from collections import defaultdict
 import argparse
+import os
 import textwrap
-from .gen_autograd import load_aten_declarations
 
-
-def gen_annotated(aten_path, out, template_path):
-    declarations = load_aten_declarations(aten_path)
-    annotated_args = []
-    for func in recurse_dict(get_py_torch_functions(declarations)):
-        annotated_args.append(process_func("torch._C._VariableFunctions", func))
-
-    for func in recurse_dict(get_py_nn_functions(declarations)):
-        annotated_args.append(process_func("torch._C._nn", func))
-
-    for func in recurse_dict(get_py_variable_methods(declarations)):
-        annotated_args.append(process_func("torch.Tensor", func))
-
-    annotated_args = textwrap.indent("\n".join(annotated_args), "    ")
-    env = {"annotated_args": annotated_args}
-    PY_ANNOTATED_ARGS = CodeTemplate.from_file(template_path + '/templates/annotated_fn_args.py')
-    write(out, 'annotated_fn_args.py', PY_ANNOTATED_ARGS, env)
-
-
-def process_func(namespace, func):
-    args = func["arguments"]
-    out_args = []
-    for arg in args:
-        if 'default' in arg or arg.get('kwarg_only', False) or arg.get('output', False):
+from typing import Dict, List, Any
+
+from tools.codegen.gen import with_native_function, parse_native_yaml, FileManager
+from tools.codegen.model import *
+import tools.codegen.api.python as python
+from .gen_python_functions import should_generate_py_binding, is_py_torch_function, is_py_nn_function, is_py_variable_method
+
+def gen_annotated(native_yaml_path: str, out: str, autograd_dir: str) -> None:
+    native_functions = parse_native_yaml(native_yaml_path)
+    mappings = (
+        (is_py_torch_function, 'torch._C._VariableFunctions'),
+        (is_py_nn_function, 'torch._C._nn'),
+        (is_py_variable_method, 'torch.Tensor'),
+    )
+    annotated_args: List[str] = []
+    for pred, namespace in mappings:
+        groups = defaultdict(list)
```

Comment:
in strict mypy you gonna need a type annotation here

Reply:
@ezyang hmm, I already added this file to mypy-strict.ini in the sequent PR on this stack and it gave clean result:
```
mypy --config mypy-strict.ini 
Success: no issues found in 11 source files
```
Is there any problem with the config we should fix?

Reply:
Hmm, this will be good to figure out.  When I apply this diff:

```
diff --git a/tools/codegen/gen.py b/tools/codegen/gen.py
index 5324246d6c..5cd4a6ffa8 100644
--- a/tools/codegen/gen.py
+++ b/tools/codegen/gen.py
@@ -1191,5 +1191,7 @@ def main() -> None:
         core_fm.write_outputs(f"{options.output_dependencies}-core")
         cuda_fm.write_outputs(f"{options.output_dependencies}-cuda")
 
+x = defaultdict(list)
+
 if __name__ == '__main__':
     main()
```

and run type checking I get

```
(/home/ezyang/local/pytorch-tmp-env) [ezyang@devvm066.ash0 ~/local/pytorch-tmp] mypy --config mypy-strict.ini 
tools/codegen/gen.py:1194:1: error: Need type annotation for 'x'
Found 1 error in 1 file (checked 7 source files)
```